### ISSUE TITLE
Audio fingerprinting

### ISSUE THREAD

**Issue by danielcra** [dillbyrne/random-agent-spoofer#429]
Hi,

Could you think about ways to add protecting against audio fingerprinting in future releases?
As I understand, it works in a similar way as Canvas fingerprinting, just using audio devices.
See here for more details:
http://thehackernews.com/2016/05/audio-fingerprint.html

Thanks

**Comment by forkoz** [comment]
Yes please, there is no way to stop this at the moment. I'm going to come out and say it... I hate HTML5 and wish it would die a fiery death.

**Comment by dillbyrne** [comment]
This is very much in the scope of RAS so I will try to get in the version after next

**Comment by one-two-three** [comment]
Other than disabling JavaScript, which isn't always an option since the modern web relies heavily on it, there's currently no way of defending against this type of fingerprinting so it should be of high priority. If RAS is the first to do this, it could also help attract more users.

**Comment by Wannabe-Programmer** [comment]
@one-two-three Completely agree with this.

Fantastic way to gain more attraction and development to the project if RAS is the first to implement such protection.

**Comment by forkoz** [comment]
it seems like i can"t stop auto loading of audio content at all which if i'm not wrong is why it works so well. i see flashblocked video's sound load still. thanks HTML5

**Comment by Wannabe-Programmer** [comment]
Check out the test here: https://audiofingerprint.openwpm.com/

**Comment by SHHSSH** [comment]
Option to disable network cache (forced by default to fix issue #121 looking for a work around)

^ Is this basically a DNS flusher?

**Comment by dillbyrne** [comment]
> Option to disable network cache (forced by default to fix issue #121 looking for a work around)

> ^ Is this basically a DNS flusher?

No it was a workaround to fix a bug where the canvas was not being disabled when a user navigated back to a page which was already loaded.  I have rewritten that code so the fix is no, longer needed.

**Comment by dillbyrne** [comment]
@Wannabe-Programmer Thanks for the update. I have to update some more profiles and I'll test and issue a release, to have a working RAS release. I'll try to tackle the audio fingerprinting and fix the date code in the release after that.

**Comment by Wannabe-Programmer** [comment]
According to this [tread](https://trac.torproject.org/projects/tor/ticket/13017), some AudioContext attributes stay between common values. 

For most users the `"ac-sampleRate"` should be one of the following -  44100Hz, 48000Hz, 96000Hz and the `"ac-maxChannelCount"` should be either 0, 2, 32 or 10000 using Firefox. 

The tests were done here: https://audiofingerprint.openwpm.com

**Comment by Rudd-O** [comment]
What do you mean by "stay between common values"?

**Comment by Wannabe-Programmer** [comment]
@Rudd-O I mean that some properties of the [AudioContext](https://developer.mozilla.org/en/docs/Web/API/AudioContext) API stay the same for most users of Firefox - or at least within a small range. This essentially means that some of the outputs are not unique. Therefore, if we plan on spoofing these values, it shouldn't not be completely random. 

Obviously, more research needs to be done before anyone can actually do anything.

**Comment by dillbyrne** [comment]
I have added the option to block it and whitelist it for now. #500 
When I get some time I'll look into spoofing the API. If someone would like to have a go feel free.
